modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/accountmanager/PocketCastsAuthenticatorService.kt

```diff
@@ -10,14 +10,20 @@ import javax.inject.Inject
 @AndroidEntryPoint
 class PocketCastsAuthenticatorService : Service() {
     @Inject lateinit var syncManager: SyncManager
-    lateinit var authenticator: PocketCastsAccountAuthenticator
+
+    private var authenticator: PocketCastsAccountAuthenticator? = null
 
     override fun onCreate() {
         super.onCreate()
         authenticator = PocketCastsAccountAuthenticator(this, syncManager)
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+        authenticator = null
+    }
+
     override fun onBind(intent: Intent?): IBinder? {
-        return authenticator.iBinder
+        return authenticator?.iBinder
```

Comment:
This fixes the following leak.

<img width="422" alt="Screenshot 2023-09-22 at 4 48 26 pm" src="https://github.com/Automattic/pocket-casts-android/assets/308331/fdbf7c91-0365-48f6-9edf-d8c5871a4eb5">